src/devices/bus/hp9825_optroms/hp9825_optrom.cpp

```diff
@@ -0,0 +1,184 @@
+// license:BSD-3-Clause
+// copyright-holders: F. Ulivi
+/*********************************************************************
+
+    hp9825_optrom.cpp
+
+    Optional ROMs for HP9825 systems
+
+*********************************************************************/
+
+#include "emu.h"
+#include "hp9825_optrom.h"
+#include "softlist.h"
+
+// Debugging
+//#define VERBOSE 1
+#include "logmacro.h"
+
+DEFINE_DEVICE_TYPE(HP9825_OPTROM_CART, hp9825_optrom_cart_device, "hp9825_optrom_cart", "HP9825 optional ROM cartridge")
+DEFINE_DEVICE_TYPE(HP9825_OPTROM_SLOT, hp9825_optrom_slot_device, "hp9825_optrom_slot", "HP9825 optional ROM slot")
+
+struct optrom_region_t {
+	offs_t m_start;
+	offs_t m_size;
+	const char *m_tag;
+};
+
+constexpr std::array<struct optrom_region_t , 8> region_tab =
+	{
+	 0x3000 , 0x400 , "rom3000" ,
+	 0x3400 , 0x400 , "rom3400" ,
+	 0x3800 , 0x400 , "rom3800" ,
+	 0x3c00 , 0x400 , "rom3c00" ,
+	 0x4000 , 0x400 , "rom4000" ,
+	 0x4400 , 0x800 , "rom4400" ,
+	 0x4c00 , 0x400 , "rom4c00" ,
+	 0x5c00 ,0x2000 , "rom5c00"
+	};
```

Comment:
You need double braces `{{` and `}}` for `std::array` initialisation to satisfy stricter compilers like GCC5 (the outer ones wrap the initialisation, the inner ones wrap the initialiser list.  Also, shouldn't you have braces around the arguments for each item?  Like `= {{ {0x3000, 0x400, "rom3000"}, {0x3400, 0x400, "rom3400"}, ... }};` but not necessarily all on one line.